internal/dataplane/kong_client.go

```diff
@@ -439,6 +452,50 @@ func (c *KongClient) Update(ctx context.Context) error {
 	return nil
 }
 
+func (c *KongClient) FetchLastGoodConfiguration(ctx context.Context) error {
+	if c.lastValidKongState == nil {
+		gatewayClients := c.clientsProvider.GatewayClients()
+		c.logger.Debugf("sending configuration to %d gateway clients", len(gatewayClients))
+
+		type kongStateWithHash struct {
+			kongState *kongstate.KongState
+			hash      string
+		}
+
+		var goodKongState *kongStateWithHash
+		_, err := iter.MapErr(gatewayClients, func(client **adminapi.Client) (*kongStateWithHash, error) {
+			rs, err := dump.Get(ctx, (*client).AdminAPIClient(), dump.Config{})
+			if err != nil {
+				return nil, err
+			}
+			ks := kongstate.KongRawStateToKongState(rs)
+			status, err := c.configGetter.GetCurrentStatus(ctx, (*client).AdminAPIClient())
+			if err != nil {
+				return nil, err
+			}
+			(*client).SetLastConfigSHA([]byte(status.ConfigurationHash))
```

Comment:
I think we should not use the `ConfigurationHash` from the status response in this context. The hash we store locally for every client is a hash generated with `deckgen.GenerateSHA` and it will not match the one generated on the Gateway side that's returned in the status response.

It's also a little unintuitive to my taste why we would like to set this in a function that's rather dedicated to reading than writing (`FetchLastGoodConfiguration`). I suggest dropping the `SetLastConfigSHA` call from here as I believe this will happen anyway in `sendToClient`.